product/opni/components/Configuration.vue

```diff
@@ -1,19 +1,140 @@
 <script>
+import jsyaml from 'js-yaml';
+import YamlEditor from '@/components/YamlEditor';
+import Loading from '@/components/Loading';
+import { exceptionToErrorsArray } from '@/utils/error';
+import AsyncButton from '@/components/AsyncButton';
+import Banner from '@/components/Banner';
+import { getGatewayConfig, updateGatewayConfig } from '~/product/opni/utils/requests';
+
 export default {
-  components: {},
+  components: {
+    YamlEditor,
+    AsyncButton,
+    Loading,
+    Banner
+  },
+
+  async fetch() {
+    await this.load();
+  },
 
   data() {
-    return { clusters: [] };
+    return {
+      loading:          false,
+      documents:        [],
+      editorContents:   '',
+      error:           '',
+    };
   },
+
+  methods: {
+    async load() {
+      try {
+        this.loading = true;
+        this.$set(this, 'documents', await getGatewayConfig(this));
+        let contents = '';
+
+        this.documents.forEach((doc, i) => {
+          contents += doc.yaml;
+          if (i < this.documents.length - 1) {
+            if (contents[contents.length - 1] !== '\n') {
+              contents += '\n';
+            }
+            contents += '---\n';
+          }
+        });
+        this.editorContents = contents;
+      } finally {
+        this.loading = false;
+      }
+    },
+    async save(buttonCallback) {
+      try {
+        const documents = jsyaml.loadAll(this.editorContents);
+        const jsonDocuments = [];
+
+        documents.forEach((doc) => {
+          jsonDocuments.push(JSON.stringify(doc));
+        });
+        await updateGatewayConfig(jsonDocuments);
+      } catch (err) {
+        buttonCallback(false);
+        this.$set(this, 'error', exceptionToErrorsArray(err).join('; '));
+      } finally {
+        buttonCallback(!this.error);
+      }
+    },
+    async reset() {
+      this.$set(this, 'error', '');
+      await this.load();
+    },
+    onInput(yaml) {
+      this.editorContents = yaml;
+    }
+  }
 };
 </script>
 <template>
-  <div>
-    <div class="bar">
-      Configuration
+  <Loading v-if="loading || $fetchState.pending" />
+  <div v-else>
+    <header>
+      <div class="title">
+        <h1>Configuration</h1>
+      </div>
+      <div slot="actions" class="row actions-container">
+        <!-- <div v-if="errors.length" class="error-message pr-10">
+          {{ errors.join('; ') }}
+        </div> -->
+        <AsyncButton
+          ref="saveBtn"
```

Comment:
I might be missing it but I'm not seeing ref used in the code. You should be able to remove.